wolfcrypt/src/blake2b.c

```diff
@@ -356,7 +356,7 @@ int blake2b_final( blake2b_state *S, byte *out, byte outlen )
     }
 
     S->buflen -= BLAKE2B_BLOCKBYTES;
-    XMEMCPY( S->buf, S->buf + BLAKE2B_BLOCKBYTES, (wolfssl_word)S->buflen );
+    XMEMMOVE( S->buf, S->buf + BLAKE2B_BLOCKBYTES, (wolfssl_word)S->buflen );
```

Comment:
XMEMOVE is not required as buflen is meant to be less BLAKE2B_BLOCKBYTES * 2.

Put a check in that ensures buflen is not invalid.
It is a safety check (in case someone changes the implementation) and will let Coverity know that XMEMMOVE isn't needed.